Js testing

Thunae opened the pull request:
Testing our Javascript

Thunae committed:
Base of Jasmine Tests

Thunae committed:
Base of Jasmine Tests

JS Testing of User Location and Address

Tests for the query functions

Remove let x, make an arrow function, fix typos

Fix mistake with user tests, localize location variable

Thunae committed:
JS Testing of User Location and Address

Thunae committed:
Tests for the query functions

lcatherine commented on src/main/java/com/google/sps/data/Response.java:
```diff
@@ -44,8 +44,7 @@ public String getStatus() {
             int randIdx = (results.size() > 1) ? new Random().nextInt(results.size() - 1) : 0;
-            pick = results.get(randIdx);
-            results.remove(randIdx);
+            this.pick = results.get(randIdx); results.remove(randIdx);
```
formatting

Sorry I did have the formatting correct in the issue I opened

Thunae replied:
done

lcatherine commented on src/main/webapp/js/query.js:
```diff
@@ -15,7 +15,7 @@
     RESTAURANT QUERY AND RE-ROLL
 =========================*/
-$("#randomize-form").submit(function(event) {
+$("#randomize-form").submit(function(event){
```
arrow function preferred

Thunae replied:
done

lcatherine commented on src/main/webapp/js/query.spec.js:
```diff
@@ -0,0 +1,333 @@
+        spyOn(window, 'resultsPage');
+    });
+
+    it("Tese the query funciton works properly", async () => {
```
I think you meant test

Thunae replied:
fixed

lcatherine commented on src/main/webapp/js/query.js:
```diff
@@ -233,14 +235,15 @@ $("input, textarea").blur(function() {
             document.getElementById("pick").innerText = name;
             document.getElementById("rating").innerText = rating;
             loadImage(photoUrl);
+            return "";
```
will returning "" cause issues? Just want to make sure...

Thunae replied:
removed

lyze commented on src/main/webapp/js/query.spec.js:
```diff
@@ -0,0 +1,333 @@
+
+    it("Tese the query funciton works properly", async () => {
+        spyOn(window, 'fetch').and.returnValue(Promise.resolve(htmlResponse));
+        let x = await query("string");
```
do we need `let x`? it seems like we don't use this in any of the tests

Thunae replied:
removed x

lyze commented on src/main/webapp/js/query.spec.js:
```diff
@@ -0,0 +1,333 @@
+    let htmlReponse;
+
+    beforeEach(() => {
+        htmlResponse = new Response('{\"pick\":{\"name\":\"Chipotle\",\"rating\":\"4.5\",\"photos\":[\"photo1\",\"photo2\"]},\"results\":[{\"name\":\"Qdoba\",\"rating\":\"4\",\"photos\":[\"photo1\",\"photo2\"]}],\"status\":\"OK\"}');
```
would a backtick string literal be easier? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

also apply below

Thunae replied:
response was created with a JSON generator so it wasn't too hard

Thunae committed:
Minor Fixes to make master function correctly (#40)

* Minor Fixes to make master function correctly

* Spacing

* Reimplement arrow function, change jquery selector

Thunae committed:
Remove let x, make an arrow function, fix typos

Thunae committed:
Add Map to the resultsPage (#36)

* Add Map to the resultsPage

* Formatting updates, arrow function

Thunae committed:
Advanced algorithm (#37)

* Add weighted algorithm chooseRestaurant to queryservlet

* Fix variable names

* Align function names

* Make naming consistent, change Assert.assert

* Addition of the Past Searches page, updates to the SearchServlet, and addition of the FeedbackServlet. This fetches the user's past searches, formats them on the html page, and adds a modal through which the user can submit feedback on the restaurant.  (#33)

* fix merge conflicts, add changes

* Past searches cards are added

* restaurant name carries through to feedback form

* feedback servlet form submission completed, servlet is fully functional

* UI updates + remove commented out lines, add extra line at end of file

* feedback servlet changes imported

* start adding comments, fix merge conflicts

* imported changes from updatedSearches branch, add comments

* fix order of import statement, extra comment signs, and unnecessary comments

* add spaces, fix function syntax, delete duplicated file

Co-authored-by: ryan <ryangudal@gmail.com>

* Secrets Manager Implementation (#35)

* Merge

* Merge

* Merge

* Restaurant Images routed through servlet to protect API key

* Secret Manager Class, Route Geolocation and Images through servlets

* Address PR comments: make access secret a singleton

* IOException caught

* Style header in caps

* Make AccessSecret member strings final

* Remove debug statements, add imports

* Address PR comments, add spacing, rework expressions

* Remove debug statements from secrets

* spacing fixes

* addition of account page updates to the correct branch (cherrypickBranch) of past searches updates  (#34)

* elements added to account page

* imported changes from updatedSearches branch, add comments

* UI update

* removed target and old (unused) files from git

* fix issue from merge, change syntax of async function

* fix function syntax

* Minor Fixes to make master function correctly (#40)

* Minor Fixes to make master function correctly

* Spacing

* Reimplement arrow function, change jquery selector

* Add Map to the resultsPage (#36)

* Add Map to the resultsPage

* Formatting updates, arrow function

* Add weighted algorithm chooseRestaurant to queryservlet

Fix variable names

Align function names

Make naming consistent, change Assert.assert

spacing fixes

Add URL encoder, fix spacing

* Bug fix: add user so call is defined

Co-authored-by: evd123 <46075902+evd123@users.noreply.github.com>

lyze commented on src/main/webapp/js/query.spec.js:
```diff
@@ -0,0 +1,333 @@
+/*=========================
+    USER SIGN-IN
+ =========================*/
+describe("Test the addUserContent Function", () => {
```
:)

Thunae replied:
whoops

lyze commented on src/main/webapp/js/query.spec.js:
```diff
@@ -0,0 +1,333 @@
+});
+
+describe("Test the convertLocation function", () => {
+    let location;
```
can we make this a local variable?

Thunae replied:
done

Thunae committed:
Java testing (#44)

* Framework for Mockito and JUnit Testing

* Clean up tests after Rebasing from master

* Add more tests for datastore and login

* Address Catherine PR Comments: remove initmocks and change imports

* Move static imports and change to camelCase

* Debugging to bring to speed with master

* Minor Fixes to make master function correctly (#40)

* Minor Fixes to make master function correctly

* Spacing

* Reimplement arrow function, change jquery selector

* Add Map to the resultsPage (#36)

* Add Map to the resultsPage

* Formatting updates, arrow function

* Framework for Mockito and JUnit Testing

Clean up tests after Rebasing from master

Add more tests for datastore and login

Address Catherine PR Comments: remove initmocks and change imports

Move static imports and change to camelCase

Debugging to bring to speed with master

* Rearrange Imports, Change test names to camelCase

* Advanced algorithm (#37)

* Add weighted algorithm chooseRestaurant to queryservlet

* Fix variable names

* Align function names

* Make naming consistent, change Assert.assert

* Addition of the Past Searches page, updates to the SearchServlet, and addition of the FeedbackServlet. This fetches the user's past searches, formats them on the html page, and adds a modal through which the user can submit feedback on the restaurant.  (#33)

* fix merge conflicts, add changes

* Past searches cards are added

* restaurant name carries through to feedback form

* feedback servlet form submission completed, servlet is fully functional

* UI updates + remove commented out lines, add extra line at end of file

* feedback servlet changes imported

* start adding comments, fix merge conflicts

* imported changes from updatedSearches branch, add comments

* fix order of import statement, extra comment signs, and unnecessary comments

* add spaces, fix function syntax, delete duplicated file

Co-authored-by: ryan <ryangudal@gmail.com>

* Secrets Manager Implementation (#35)

* Merge

* Merge

* Merge

* Restaurant Images routed through servlet to protect API key

* Secret Manager Class, Route Geolocation and Images through servlets

* Address PR comments: make access secret a singleton

* IOException caught

* Style header in caps

* Make AccessSecret member strings final

* Remove debug statements, add imports

* Address PR comments, add spacing, rework expressions

* Remove debug statements from secrets

* spacing fixes

* addition of account page updates to the correct branch (cherrypickBranch) of past searches updates  (#34)

* elements added to account page

* imported changes from updatedSearches branch, add comments

* UI update

* removed target and old (unused) files from git

* fix issue from merge, change syntax of async function

* fix function syntax

* Minor Fixes to make master function correctly (#40)

* Minor Fixes to make master function correctly

* Spacing

* Reimplement arrow function, change jquery selector

* Add Map to the resultsPage (#36)

* Add Map to the resultsPage

* Formatting updates, arrow function

* Add weighted algorithm chooseRestaurant to queryservlet

Fix variable names

Align function names

Make naming consistent, change Assert.assert

spacing fixes

Add URL encoder, fix spacing

* Bug fix: add user so call is defined

Co-authored-by: evd123 <46075902+evd123@users.noreply.github.com>

* Framework for Mockito and JUnit Testing

Clean up tests after Rebasing from master

Add more tests for datastore and login

Address Catherine PR Comments: remove initmocks and change imports

Move static imports and change to camelCase

Debugging to bring to speed with master

Framework for Mockito and JUnit Testing

Clean up tests after Rebasing from master

Add more tests for datastore and login

Debugging to bring to speed with master

Rearrange Imports, Change test names to camelCase

* Clean after Rebasing from Master

Co-authored-by: evd123 <46075902+evd123@users.noreply.github.com>

Thunae committed:
Fix mistake with user tests, localize location variable

Thunae committed:
Fix the Jasmine Tests are rebasing

Thunae committed:
Fix Jasmine Tests after rebasing with master and merging
